student/src/test/java/tds/student/services/remote/RemotePrintServiceTest.java

```diff
@@ -0,0 +1,599 @@
+package tds.student.services.remote;
+
+import TDS.Shared.Exceptions.ReturnStatusException;
+import net.logstash.logback.encoder.org.apache.commons.lang.StringUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.UUID;
+
+import tds.exam.ExamPrintRequest;
+import tds.itemrenderer.data.AccLookup;
+import tds.itemrenderer.data.ITSAttachment;
+import tds.itemrenderer.data.ITSContent;
+import tds.itemrenderer.data.ITSDocumentXml;
+import tds.student.services.abstractions.IContentService;
+import tds.student.services.abstractions.PrintService;
+import tds.student.services.data.ItemResponse;
+import tds.student.services.data.PageGroup;
+import tds.student.services.data.TestOpportunity;
+import tds.student.sql.data.OpportunityInstance;
+import tds.student.sql.data.OpportunityItem;
+import tds.student.sql.data.TestConfig;
+import tds.student.sql.repository.ExamRepository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RemotePrintServiceTest {
+  @Mock
+  private PrintService legacyPrintService;
+
+  @Mock
+  private ExamRepository mockExamRepository;
+
+  @Mock
+  private IContentService mockContentService;
+
+  private PrintService service;
+
+  @Captor
+  private ArgumentCaptor<ExamPrintRequest> examPrintRequestCaptor;
+
+  final OpportunityInstance mockOpportunityInstance = new OpportunityInstance(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID());
```

Comment:
Trust me, I wanted to. Unfortunately the version of that library for Java 7 has a dependency on a newer version of the apache.commons.lang3 tools. I tried getting it to work briefly, but it looked like that dependency was being pulled in from a parent pom. I wasn't comfortable enough upgrading to the newer version for fear of introducing a regression, but if you guys want me to give it a shot and do some testing, we can. It has saved a huge amount of time on the exam side.